File: android/src/main/kotlin/com/benjamin/horner/flutter_acs_card_reader/notifiers/DeviceNotifier.kt
package com.benjamin.horner.flutter_acs_card_reader

import io.flutter.plugin.common.MethodChannel

import javax.smartcardio.CardTerminal

class DeviceNotifier {
    fun cardTerminalToMap(cardTerminal: CardTerminal): Map<String, Any?> {
        val terminalMap = mutableMapOf<String, Any?>()
        terminalMap["name"] = cardTerminal.name
        terminalMap["isCardPresent"] = cardTerminal.isCardPresent
        return terminalMap
    }
    
    fun updateState(terminal: CardTerminal, channel: MethodChannel) {
        channel.invokeMethod("onDeviceFoundEvent", cardTerminalToMap(terminal))
    }
}

File: ios/Classes/Helpers/DeviceNotifier.swift
//
//  DeviceNotifier.swift
//  flutter_acs_card_reader
//
//  Created by Benjamin Horner on 24/11/2023.
//

import Flutter
import SmartCardIO

class DeviceNotifier {
    func cardTerminalToMap(cardTerminal: CardTerminal) throws -> [String: Any?] {
        do {
            var terminalMap = [String: Any?]()
            terminalMap["name"] = cardTerminal.name
            terminalMap["isCardPresent"] = try cardTerminal.isCardPresent()
            return terminalMap
        } catch {
            throw error
        }
    }

    func updateState(terminal: CardTerminal, channel: FlutterMethodChannel) throws {
        do {
            let terminalMap: [String: Any?] = try cardTerminalToMap(cardTerminal: terminal)
            channel.invokeMethod("onDeviceFoundEvent", arguments: terminalMap)
        } catch {
            throw error
        }
    }
}
